packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -1,16 +1,91 @@
 
 > All classes with the `Cfn` prefix in this module ([CFN Resources](https://docs.aws.amazon.com/cdk/latest/guide/constructs.html#constructs_lib)) are always stable and safe to use.
 
+![cdk-constructs: Experimental](https://img.shields.io/badge/cdk--constructs-experimental-important.svg?style=for-the-badge)
+
+> The APIs of higher level constructs in this module are experimental and under active development. They are subject to non-backward compatible changes or removal in any future version. These are not subject to the [Semantic Versioning](https://semver.org/) model and breaking changes will be announced in the release notes. This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
 ---
 <!--END STABILITY BANNER-->
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+Amazon CloudWatch Synthetics allow you to monitor your application by generating **synthetic** traffic. The traffic is produced by a **canary**: a configurable script that runs on a schedule. You configure the canary script to follow the same routes and perform the same actions as a user, which allows you to continually verify your user experience even when you don't have any traffic on your applications.
+
+## Canary
+
+To illustrate how to use a canary, assume your application defines the following endpoint:
+
+```bash
+% curl "https://api.example.com/user/books/topbook/"
+The Hitchhikers Guide to the Galaxy
+
+```
+
+The below code defines a canary that will hit the `books/topbook` endpoint every 5 minutes:
 
 ```ts
 import * as synthetics from '@aws-cdk/aws-synthetics';
+
+const canary = new synthetics.Canary(this, 'MyCanary', {
+  frequency: Duration.minutes(5),
+  test: synthetics.Test.custom({
+    code: synthetics.Code.fromInline(`const https = require('https');
+      var synthetics = require('Synthetics');
+      const log = require('SyntheticsLogger');
+  
+      exports.handler = async function () {
+        const requestOptions = {"hostname":"api.example.com","method":"GET","path":"/user/books/topbook/","port":443}
+        let req = https.request(requestOptions);
+        req.on('response', (res) => {
+          log.info()
+        });
+      }`),
+    handler: 'index.handler',
+  }),
+});
+```
+
+The canary will automatically produce a CloudWatch Dashboard:
+
+![UI Screenshot](images/ui-screenshot.png)
+
+### Canary Test Property
+
+The `test` property represents the test that the canary executes. You can call `Test.custom()`, which will allow you to specify a custom script and handler for the canary. To specify the script in the `code` property, use the static method `code.fromInline()`.
+
+```ts
+const canary = new Canary(this, 'MyCanary', {
+  test: Test.custom({
+    code: Code.fromInline('exports.handler = async () => {\nconsole.log(\'hello world\');\n};'),
+    handler: 'index.handler',
+  }),
+});
```

Comment:
From what I see, the `Test.custom` method accepts a `scope` as the first argument, so this would require:

```suggestion
const canary = new Canary(this, 'MyCanary', {
  test: Test.custom(this, {
    code: Code.fromInline('exports.handler = async () => {\nconsole.log(\'hello world\');\n};'),
    handler: 'index.handler',
  }),
});
```

Which creates a redundant specification of `scope`.

This also is not aligned with the experience we have for lambda, for example:

```typescript
new lambda.Function(this, 'Function', {
  code: lambda.Code.fromAsset('/path/to/asset')
});
```

Here we don't have this requirement to pass `scope` to `Code.fromAsset`. 

See my comments below on how this should be implemented.